### PR TITLE
Feat/show deal command 2761

### LINKED ISSUE #2761: Add a show deal command
### Description

As a follow up to #2729, we should add a command for inspecting individual storage deals as @ZenGround0 suggested. We can put this command in `go-filecoin show deal` and have it accept a Cid as an argument. It can use the `DealGet` plumbing method to fetch the deal then print it in a nice human readable format with more detail than either of the `list-deal` commands.

### Acceptance criteria

1. A `go-filecoin show deal` command exists
2. It's tested

### Risks + pitfalls

What the output should look like and what should be included in it is open ended. Just go with whatever seems reasonable.

### Where to begin

Check out `DealGet` in `porcelain/storagedeals.go`

### PR DESCRIPTION
**Pull request by shannonwells**
Closes #2761 

Add a command: `go-filecoin show deal <cid>` command.  If the CID can't be found it says the CID can't be found. If the CID is found, it shows this output:

```
$ go-filecoin show deal zDPWYqFCrfNeYSgaPYaYNzprPBfsy88Eoqn4aGk7Lcg4NdbMy1XF
        CID: zDPWYqFCrfNeYSgaPYaYNzprPBfsy88Eoqn4aGk7Lcg4NdbMy1XF
        State: staged
        Miner: t2lluak7lghwv56n5vyxfxd6wd7y6t5jfpgppoi2q
        Duration: 5 blocks
        Size: 12 bytes
        Total Price: 1200 FIL
```

Also I moved the show command with its `block` subcommand into *show.go* because at first I didn't see it, and did not know it was in *block.go*.  This is in line with the filenames for the other commands.

### CODE REVIEW

File: commands/show_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		clientDaemon.WaitForMessageRequireSuccess(addAskCid)
+		dataCid := clientDaemon.RunWithStdin(strings.NewReader("HODLHODLHODL"), "client", "import").ReadStdoutTrimNewlines()
+
+		proposeDealOutput := clientDaemon.RunSuccess("client", "propose-storage-deal", fixtures.TestMiners[0], dataCid, "0", "5").ReadStdoutTrimNewlines()
+
+		splitOnSpace := strings.Split(proposeDealOutput, " ")
+
+		dealCid := splitOnSpace[len(splitOnSpace)-1]
+
+		showdeal := minerDaemon.RunSuccess("show", "deal", dealCid).ReadStdoutTrimNewlines()
```
Comment (ZenGround0) [inline commands/show_test.go]: (Disclaimer this comment doesn't imply any more work to this PR)
It's unfortunate it takes us setting up a small network to get a deal datastructure.  I think a long term goal is to have tooling exposing data structure construction, signing and adding to the node.  In that better world this test setup would consist of creating two keys / wallets, deal construction would happen with a few commands and loading the deal to the node.

**Comment by rosalinekarr** [comment]
Actually, one note: I believe @acruikshank was hoping to include a deals payment vouchers in the output from this command. That might be a lot of work beyond what's already here, so I'd say just add an issue for including that in the future and we can address it later.

File: commands/show.go [hunk 2]
```diff
@@ -6,9 +6,10 @@ import (
 	"strconv"
 
 	"github.com/ipfs/go-cid"
-	"github.com/ipfs/go-ipfs-cmdkit"
-	"github.com/ipfs/go-ipfs-cmds"
+	cmdkit "github.com/ipfs/go-ipfs-cmdkit"
```
[the remark this answers is not in the document]
Comment (shannonwells) [reply commands/show.go]: guh.  will fix

**Comment by laser** [comment]
@shannonwells - Looks like you've got the approvals you need to merge - but one of the functional tests failed the build.

File: commands/show_test.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		// ensure that the JSON we received from block get conforms to schema
+
+		requireSchemaConformance(t, []byte(blockGetLine), "filecoin_block")
+	})
+}
+
+func TestShowDeal(t *testing.T) {
+	tf.IntegrationTest(t)
+
+	minerDaemon := th.NewDaemon(t,
```
[the remark this answers is not in the document]
Comment (shannonwells) [reply commands/show_test.go]: @ZenGround0 I spent quite a bit of time trying to get one to work, but there seem to be some timing bugs. I am going to file a couple of issues for this, including converting this test to FAST.

**Comment by shannonwells** [comment]
> Actually, one note: I believe @acruikshank was hoping to include a deals payment vouchers in the output from this command. That might be a lot of work beyond what's already here, so maybe just add an issue for including that in the future and we can address it later.

I'll file a follow-up since it's right before summit. this one has been hanging out too long.

**Pull request merged**